client/state/stats/emails/utils.js

```diff
@@ -218,12 +218,13 @@ export function parseEmailListData( list ) {
  * Return link data in a format used by lists for email stats. The fields array is matched to
  * the data in a single object.
  *
- * @param {Array} links - the array of links for the given data
+ * @param {Array} internalLinks - the array of links for the given data
+ * @param {Array} userContentLinks - the array of user content links for the given data
  * @returns {Array|null} - Array of data objects
  */
-export function parseEmailLinksData( links ) {
-	if ( ! links ) {
-		return null;
+export function parseEmailLinksData( internalLinks, userContentLinks ) {
+	if ( ! Array.isArray( internalLinks ) || ! Array.isArray( userContentLinks ) ) {
```

Comment:
According to the call to this function:
```js
		links: parseEmailLinksData( stats.links?.data, stats[ 'user-content-links' ]?.data ),
```
The second parameter could be empty. In that case, the internalLinks won't be parsed.

Reply:
You're 100% right, I'll refactor